src/OpenTelemetry.Contrib.Extensions.AWSXRay/Instrumentation/AWSTracingPipelineHandler.cs

```diff
@@ -0,0 +1,249 @@
+        private static readonly Action<IDictionary<string, string>, string, string> Setter = (carrier, name, value) =>
+        {
+            carrier[name] = value;
+        };
+
+        private static readonly Dictionary<string, string> ServiceParameterMap = new Dictionary<string, string>()
+        {
+            { "DynamoDBv2", "TableName" },
+            { "SQS", "QueueUrl" },
+        };
+
+        private static readonly Dictionary<string, string> ParameterAttributeMap = new Dictionary<string, string>()
+        {
+            { "TableName", AWSSemanticConventions.AttributeAWSDynamoTableName },
+            { "QueueUrl", AWSSemanticConventions.AttributeAWSSQSQueueUrl },
+        };
+
+        private static readonly ActivitySource AWSSDKActivitySource = new ActivitySource(ActivitySourceName);
+
+        public override void InvokeSync(IExecutionContext executionContext)
+        {
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                base.InvokeSync(executionContext);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+        }
+
+        public override async Task<T> InvokeAsync<T>(IExecutionContext executionContext)
+        {
+            T ret = null;
+
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                ret = await base.InvokeAsync<T>(executionContext).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+
+            return ret;
+        }
+
+        private Activity ProcessBeginRequest(IExecutionContext executionContext)
+        {
+            Activity activity = null;
+
+            var requestContext = executionContext.RequestContext;
+            var service = this.GetAWSServiceName(requestContext);
+            var operation = this.GetAWSOperationName(requestContext);
+
+            activity = AWSSDKActivitySource.StartActivity(service + "." + operation, ActivityKind.Client);
+
+            if (activity == null)
+            {
+                return null;
+            }
+
+            activity.SetTag(AWSSemanticConventions.AttributeAWSServiceName, service);
```

Comment:
Typically we would check `if (activity.IsAllDataRequested)` and only add tags if data was requested. That would be the case where span was not sampled.

Reply:
My understanding is that the activity created by `activity = AWSSDKActivitySource.StartActivity(service + "." + operation, ActivityKind.Client);` will be null if it was not sampled. That's why I'm checking for nullness before any operation on the activity. Is it not true?

Reply:
@srpash

> Is it not true?

Not true 😄 Check out:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/da4d352a198a89200fcec74bd4a20838aec45e02/src/OpenTelemetry/Trace/TracerProviderSdk.cs#L234-L239

A `Drop` sample decision results in `ActivitySamplingResult.PropagationData`. Activity is created so that context still flows and is propagated, but because the span is not sampled, we shouldn't add data to it.

Reply:
Oh I see! I'll add the `activity.IsAllDataRequested` check before populating tags. Thanks for clarifying. 😄

Reply:
Added!